src/codegen/generate.ts

```diff
@@ -588,6 +588,20 @@ export default class ApiGenerator {
       // items -> array
       return factory.createArrayTypeNode(this.getTypeFromSchema(schema.items));
     }
+    if ("prefixItems" in schema) {
+      // prefixItems -> typed tuple
+      // Because we are mixing OpenAPI 3.0 and OpenAPI 3.1 types prefixItems can be defined without items
+      // For some reason typescript assumes schema to be NonArraySchemaObject after the above if statement so we cast it back
+      // Also the definition of prefixItems in openapi-types seems to be wrong so we cast again
+      const prefixItems = (schema as unknown as OpenAPIV3.ArraySchemaObject)
+        .prefixItems as unknown as (
+        | OpenAPIV3.SchemaObject
+        | OpenAPIV3.ReferenceObject
+      )[];
```

Comment:
On my end this works fine:

```ts
const prefixItems = schema.prefixItems as (
  | OpenAPIV3.SchemaObject
  | OpenAPIV3.ReferenceObject
)[];
```

Ideally we'd have typeguards here but I haven't found existing implementations for a `isSchemaObject` and `isReferenceObject` and I'd not bother implementing those ourselves just for this case.

Reply:
great solution 👍